lib/AvirWrapper.php

```diff
@@ -144,6 +137,60 @@ function () use ($scanner, $path) {
 		}
 		return $stream;
 	}
+
+	/**
+	 * @param AbstractScanner $scanner
+	 * @param string $path
+	 * @param bool $shouldDelete
+	 * @throws InvalidContentException
+	 */
+	private function onScanComplete($scanner, $path, $shouldDelete){
+		$status = $scanner->completeAsyncScan();
+		if (intval($status->getNumericStatus()) === \OCA\Files_Antivirus\Status::SCANRESULT_INFECTED) {
+			$owner = $this->getOwner($path);
+
+			$this->logger->warning(
+				'Infected file deleted. ' . $status->getDetails()
+				. ' Account: ' . $owner . ' Path: ' . $path,
+				['app' => 'files_antivirus']
+			);
+
+			\OC::$server->getActivityManager()->publishActivity(
+				'files_antivirus',
+				Activity::SUBJECT_VIRUS_DETECTED,
+				[$path, $status->getDetails()],
+				Activity::MESSAGE_FILE_DELETED,
+				[],
+				$path,
+				'',
+				$owner,
+				Activity::TYPE_VIRUS_DETECTED,
+				Activity::PRIORITY_HIGH
+			);
+
+			if ($shouldDelete){
+				//prevent from going to trashbin
+				if (App::isEnabled('files_trashbin')) {
+					\OCA\Files_Trashbin\Storage::preRenameHook([
```

Comment:
:see-no-evil: (I know, old code)

Reply:
@VicDeo can you make a ticket to look into ways to get rid of this eventually ? I hope this isn't causing side effects in some scenarios.

Reply:
@PVince81 https://github.com/owncloud/files_antivirus/issues/199